### PR TITLE
[TA] Add dynamic classification feature

### PR DESCRIPTION
**Pull request by mssfang**
# Description
fixes: https://github.com/Azure/azure-sdk-for-java/issues/30943
Add the dynamic classification new feature.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,12 @@
 - Added the Abstractive Summarization feature and related models: `AbstractSummaryAction`, `AbstractSummaryActionResult`,
  `AbstractSummaryResultCollection`, `AbstractSummaryResult`, `AbstractiveSummary`, and `SummaryContext`. 
   Access the feature through the `beginAnalyzeActions` API.
+- Added the dynamic text classification on documents without needing to train a model. The feature can be used by calling:
+    - `DynamicClassifyDocumentResultCollection dynamicClassificationBatch(Iterable<String> documents, String language, DynamicClassificationOptions options)`    
+    - `Response<DynamicClassifyDocumentResultCollection> dynamicClassificationBatchWithResponse(Iterable<TextDocumentInput> documents, DynamicClassificationOptions options, Context context)`
+    - `Mono<DynamicClassifyDocumentResultCollection> dynamicClassificationBatch(Iterable<String> documents, String language, DynamicClassificationOptions options)`
+    - `Mono<Response<DynamicClassifyDocumentResultCollection>> dynamicClassificationBatchWithResponse(Iterable<TextDocumentInput> documents, DynamicClassificationOptions options)`.
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/CHANGELOG.md:28]: Does Java normally include all the overloads present in the changelog?

File: sdk/textanalytics/azure-ai-textanalytics/README.md [hunk 2]
```diff
@@ -544,8 +546,9 @@ set of documents. Currently, the supported features are:
 - Custom Entity Recognition (API version 2022-05-01 and newer)
 - Custom Single-Label Classification (API version 2022-05-01 and newer)
 - Custom Multi-Label Classification (API version 2022-05-01 and newer)
-- Extractive Text Summarization (API version 2022-10-01-preview and newer)
+- Dynamic Text Classification (API version 2022-10-01-preview and newer)
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/README.md:549]: Dynamic classification isn't supported as an action

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java [hunk 3]
```diff
@@ -1577,6 +1579,141 @@ public Response<AnalyzeSentimentResultCollection> analyzeSentimentBatchWithRespo
...
+     *         System.out.printf&#40;&quot;&#92;tCategory: %s, confidence score: %f.%n&quot;,
+     *             classification.getCategory&#40;&#41;, classification.getConfidenceScore&#40;&#41;&#41;;
+     *     &#125;
+     * &#125;&#41;;
+     * </pre>
+     * <!-- end Client.dynamicClassificationBatch#Iterable-String-DynamicClassificationOptions -->
+     *
+     * @param documents A list of documents to be analyzed.
+     * For text length limits, maximum batch size, and supported text encoding, see
+     * <a href="https://docs.microsoft.com/azure/cognitive-services/text-analytics/overview#data-limits">data limits</a>.
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java:1616]: above you give an aka link that's shorter. Should this link be updated too? https://aka.ms/azsdk/textanalytics/data-limits

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/models/ClassificationDocumentResult.java [hunk 4]
```diff
@@ -14,7 +14,7 @@ public class ClassificationDocumentResult extends DocumentResult {
     /*
      * The class property.
      */
-    @JsonProperty(value = "class", required = true)
+    @JsonProperty(value = "classifications", required = true)
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/models/ClassificationDocumentResult.java:17]: if the service deploys a fix which changes this to start correctly returning `class`, will the java code still work?

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/DynamicClassificationOptions.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+     * @return The value of {@code categories}.
+     */
+    public Iterable<String> getCategories() {
+        return categories;
+    }
+
+    /**
+     * Sets the value of {@code categories}.
+     *
+     * @param categories the
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/DynamicClassificationOptions.java:75]: this sentence appears to trail off

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/DynamicClassificationBatchDocuments.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        // Response's status code
+        System.out.printf("Status code of request response: %d%n", dynamicClassificationResponse.getStatusCode());
+        DynamicClassifyDocumentResultCollection resultCollection = dynamicClassificationResponse.getValue();
+
+        // Model version
+        System.out.printf("Results of \"Dynamic Classification\" Model, version: %s%n", resultCollection.getModelVersion());
+
+        // Batch statistics
+        TextDocumentBatchStatistics batchStatistics = resultCollection.getStatistics();
+        System.out.printf("Documents statistics: document count = %s, erroneous document count = %s, transaction count = %s, valid document count = %s.%n",
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/DynamicClassificationBatchDocuments.java:63]: should these be strings?

File: sdk/textanalytics/azure-ai-textanalytics/CHANGELOG.md (hunk 1 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/CHANGELOG.md:28]: There is no standard for it. I could show the max-overload method name only and mention other overload methods.

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/models/ClassificationDocumentResult.java (hunk 4 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/models/ClassificationDocumentResult.java:17]: it won't for sure. Once the service corrects it, will create a PR to address it.

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/DynamicClassificationBatchDocuments.java (hunk 6 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/DynamicClassificationBatchDocuments.java:63]: it shouldn't

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/ExtractKeyPhrasesBatchStringDocuments.java [hunk 7]
```diff
@@ -47,7 +47,7 @@ public static void main(String[] args) {
 
         // Batch statistics
         TextDocumentBatchStatistics batchStatistics = keyPhrasesBatchResultCollection.getStatistics();
-        System.out.printf("Documents statistics: document count = %s, erroneous document count = %s, transaction count = %s, valid document count = %s.%n",
+        System.out.printf("Documents statistics: document count = %d, erroneous document count = %s, transaction count = %s, valid document count = %s.%n",
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/ExtractKeyPhrasesBatchStringDocuments.java:50]: only the first got updated in this one

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/RecognizeLinkedEntitiesBatchStringDocuments.java [hunk 8]
```diff
@@ -47,7 +47,7 @@ public static void main(String[] args) {
 
         // Batch statistics
         TextDocumentBatchStatistics batchStatistics = linkedEntitiesResultCollection.getStatistics();
-        System.out.printf("Documents statistics: document count = %s, erroneous document count = %s, transaction count = %s, valid document count = %s.%n",
+        System.out.printf("Documents statistics: document count = %d, erroneous document count = %s, transaction count = %s, valid document count = %s.%n",
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/batch/RecognizeLinkedEntitiesBatchStringDocuments.java:50]: same here

**Review by kristapratico — APPROVED** [review]

**Pull request merged**